### PR TITLE
Push-SDK generator-- do not merge

### PR DESCRIPTION
**Pull request by jkingoliver**
Generate scaffolding code for Push SDK (server). #67

### CODE REVIEW

File: test/unit/refresh.js [hunk 1]
```diff
@@ -22,7 +22,7 @@ var fs = require('fs');
 var format = require('util').format;
```
Comment (jkingoliver) [inline test/unit/refresh.js]: Sorry, I re-formatted this file, so it has a huge diff. I added just a single test at line 1349.

Comment (tunniclm) [reply test/unit/refresh.js]: Fixed, should be easier to see what's changed now

**Comment by tunniclm** [comment]
I'm guessing there's a dependency version problem here as we have seen the swift package manager get stuck / take a long time to resolve some dependency trees.

**Comment by jkingoliver** [comment]
Yes there's a dependency version update in the push sdk repo, see:  https://github.com/ibm-bluemix-mobile-services/bms-pushnotifications-serversdk-swift/pull/22

The update is in the develop branch and is waiting to be merged to master and released.

** Blocked until push sdk is re-released.

**Comment by jkingoliver** [comment]
Note to Jennie: update test values like here: https://github.com/IBM-Swift/generator-swiftserver/pull/211

**Comment by jkingoliver** [comment]
@rolivieri Push sdk will be re-released Friday 5/26 or early next week

**Comment by jkingoliver** [comment]
Do not merge until CloudConfiguration is updated with the PushNotificationsService.

File: refresh/templates/extensions/PushNotificationsExtension.swift [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+        let regionVar = "<%- service.region %>"
+        var region: String
+
+        regionVar.isEmpty ? (region = PushNotifications.Region.US_SOUTH) : (region = regionVar)
+
+        self.init(
+            bluemixRegion: region,
+            bluemixAppGuid: "<%- service.guid %>", 
+            bluemixAppSecret: "<%- service.secret %>" )
```
Comment (tunniclm) [inline refresh/templates/extensions/PushNotificationsExtension.swift:17]: This looks a little strange to me, I wasn't expecting to see template tags in this file. I would have expected these values to be available from the `service` parameter, is that incorrect?

I tried to check by looking up the `PushNotificationsService` in CloudConfiguration (or @jkingoliver 's fork) but I couldn't find it. I did check the service credentials Bluemix gives and it does seem to supply `appGuid`, `url`, `admin_url`, `appSecret` and `clientSecret`.

So I guess this was a mistake and these are not supposed to be templated?

Comment (jkingoliver) [reply refresh/templates/extensions/PushNotificationsExtension.swift:17]: @tunniclm I am re-working the extension and this is no longer ready for review, I'm going to close the PR and re-open when it's ready.

**Comment by jkingoliver** [comment]
I'll re-open this PR when the Cloud Config for the Push service is ready.

File: refresh/templates/extensions/PushNotificationsExtension.swift (hunk 2 above)
Comment (tunniclm) [reply refresh/templates/extensions/PushNotificationsExtension.swift:17]: ok, cool

**Pull request closed without merging**